src/rc-socket.js

```diff
@@ -68,6 +68,11 @@ var RcSocket = function (url, protocols) {
     this.unload = true;
   }.bind(this);
 
+  // Protect against network/ip changes.
+  window.addEventListener('offline', function () {
+    this.refresh();
```

Comment:
The `refresh` function closes the socket, am I correct in understanding that the next message sent will try to reopen the socket?  So `refresh` is a lazy refresh?